pkg/metrics/metrics.go

```diff
@@ -0,0 +1,190 @@
+	MetricJiraWebhookProcessFailures = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:        "compliance_audit_router_jira_webhook_process_failures",
+		Help:        "Number of Jira notification webhooks notifications that failed to be processed",
+		ConstLabels: CARPrometheusLabels},
+		[]string{"uuid", "process"},
+	)
+	// MetricJiraIssueUpdateFailures is the number of failures updating issues based on received webhook events
+	MetricJiraIssueUpdateFailures = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:        "compliance_audit_router_jira_issue_update_failures",
+		Help:        "Number of Jira issues that failed to be updated based on received webhook",
+		ConstLabels: CARPrometheusLabels},
+		[]string{"uuid", "process"},
+	)
+
+	// LDAP LOOKUP PROCESSING
+
+	// MetricLDAPLookupFailures is the number of LDAP lookups that failed
+	// Other LDAP metrics are not currently tracked, as they are not integral to the process
+	MetricLDAPLookupFailures = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:        "compliance_audit_router_ldap_lookup_failures",
+		Help:        "Number of LDAP lookups that failed",
+		ConstLabels: CARPrometheusLabels},
+		[]string{"uuid", "process"},
+	)
+
+	// HTTP RESPONSES TO CLIENTS
+
+	// MetricHTTPErrors is the number of HTTP errors returned by the application
+	MetricHTTPErrors = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:        "compliance_audit_router_http_errors",
+		Help:        "Number of HTTP errors returned by the application",
+		ConstLabels: CARPrometheusLabels},
+		[]string{"code", "uuid", "process"},
+	)
+	// MetricHTTPSuccesses is the number of HTTP successes returned by the application
+	MetricHTTPSuccesses = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name:        "compliance_audit_router_http_successes",
+		Help:        "Number of HTTP successes returned by the application",
+		ConstLabels: CARPrometheusLabels},
+		[]string{"code", "uuid", "process"},
+	)
+
+	// GENERIC APP DATA
+
+	// MetricUptime is the number of seconds the application has been running
+	MetricUptime = prometheus.NewCounter(prometheus.CounterOpts{
+		Name:        "compliance_audit_router_uptime_seconds",
+		Help:        "Number of seconds the application has been running",
+		ConstLabels: CARPrometheusLabels},
+	)
+
+	MetricsList = []prometheus.Collector{
+		MetricSplunkWebhookReceived,
+		MetricSplunkWebhookProcessFailures,
+		MetricSplunkAlertSIDReceived,
+		MetricSplunkSearchResultQueryFailures,
+		MetricComplianceEventsFound,
+		MetricComplianceEventsProcessed,
+		MetricJiraClientCreateFailures,
+		MetricJiraIssueCreated,
+		MetricJiraErrorIssuesCreated,
+		MetricJiraIssueCreateFailures,
+		MetricJiraWebhookReceived,
+		MetricJiraWebhookProcessFailures,
+		MetricJiraIssueUpdateFailures,
+		MetricLDAPLookupFailures,
+		MetricHTTPErrors,
+		MetricHTTPSuccesses,
+		MetricUptime,
+	}
+)
+
+func RecordUptime() {
+	for {
+		MetricUptime.Inc()
+		time.Sleep(time.Second)
+	}
+}
```

Comment:
Something I've seen in other metrics is exposing a metric as the Linux epoch time it started. Then, uptime can always be calculated with the PromQL:

```
time() - compliance_audit_router_started
```

I think the main advantage would be putting less load on Prometheus as well as the node this pod ends up running on

Reply:
Nice, that makes sense, and I discovered prometheus has a `process_start_time_seconds` already included, so no extra code is needed here for this.